nservicebus/sqlserver/design.md

```diff
@@ -5,72 +5,80 @@ tags:
 - SQL Server
 ---
 
-The SQL Server transport implements a message queueing mechanism on top of a Microsoft SQL Server database. It uses tables to model queues. It does not make any use of ServiceBroker, a messaging technology built into the SQL Server, mainly due to cumbersome API and difficult maintenance.
+## Queues
 
+### Primary queue
 
-## How it works?
+For each endpoint there is a single primary queue table which name matches the name of the endpoint. In scale-out scenario this single queue is shared by all instances.
 
-The SQL Server transport is a hybrid queueing system which is neither store-and-forward (like MSMQ) nor a broker (like RabbitMQ). It treats the SQL Server only as a storage infrastructure for the queues. The queueing logic itself is implemented and executed as part of the transport code running in an NServiceBus endpoint.
 
+### Secondary queues
 
-## Pros & cons
+In order for [callbacks](/nservicebus/messaging/handling-responses-on-the-client-side.md) to work in a scale-out scenario each endpoint instance has to have its own queue/table. This is necessary because callback handlers are stored in-memory in the node that did the send. The reply should be delivered to the dedicated queue, so the message can be picked up by the same node that registered the callback.
 
+Secondary queue tables have the name of the machine appended to the name of the primary queue table with `.` as separator e.g. `SomeEndpoint.MyMachine`.
 
-### Pros
+Secondary queues are enabled by default. In order to disable them, one must use the configuration API:
 
- * Nearly all Microsoft stack organizations already have SQL Server installed (no license cost) and have knowledge required to run it (no training cost)
- * Most developers are familiar with SQL Server
- * Tooling (SSMS) is great
- * Maximum throughput for any given endpoint is on par with MSMQ
- * Queues support competing consumers (multiple instances of same endpoint feeding off of same queue) so there is no need for distributor in order to scale out the processing
+snippet:sqlserver-config-disable-secondaries
 
+Secondary queues use same adaptive concurrency model to the primary queue. Secondary queues (and hence callbacks) are disabled for satellite receivers.
 
-### Cons
+### Queue table structure
 
- * No local store-and-forward mechanism meaning if SQL Server instance is down, endpoint cannot send nor receive messages
- * In centralized deployment maximum throughput applies for whole system, not individual endpoints, so if SQL Server on a given hardware can handle 2000 msg/s, each one of 10 endpoints can only receive 200 msg/s (on average).
+Following SQL DDL is used to create a table and its index for a queue:
 
+```SQL
+CREATE TABLE [schema].[queuename](
```

Comment:
As far as I know yes, so not sure if we need snippet here. In fact, I have doubts if we should even provide that information here given we have "Scripting" article.